elastic/kibana
# [Index Management] Add create enrich policy flow

## Summary

This is the first of a series of PR's that will be merged in the [enrich-policies feature branch](https://github.com/elastic/kibana/tree/feature/enrich-policies). 

This PR adds support for creating enrich policies.

Note to reviewers: 
* There is [a known bug](https://github.com/elastic/elasticsearch/issues/90171) in the `get _alias` api that will end up returning hidden indices even if wildcards=open is being used. So do expect to see that for know until ES fixes that issue.
* cit and functional tests will be added separately with https://github.com/elastic/kibana/issues/165648
* Copy and doc links will be separately addressed with https://github.com/elastic/kibana/issues/163808

##### How to test
1. Start elasticsearch with `yarn es snapshot --license=trial` and serverless search solution with `yarn start`
2. Navigate to console and execute the commands listed below in order to create the required indices and mappings for testing
3. Navigate to index management and click on enrich policies tab
4. Verify that:
  * Source indices shows the available indices in the cluster (should return indices + aliases)
  * Upload a file link next to source indices takes you to upload file in kibana
  * Form validation works
  * If multiple source indices are selected, then the match field should show the common fields from all the selected indices (if there are none, it should show an error)
  * Enrich fields should show a list of fields per index, and multiple can be selected
  * Last step allows you to either create or create and execute
  * After creation the policy should be shown in the table


<details>
<summary>Create test indices</summary>

```
PUT /test-a
{
  "mappings": {
    "properties": {
      "name": {
        "type": "text"
      },
      "email": {
        "type": "keyword"
      }
    }
  }
}
PUT /test-b
{
  "mappings": {
    "properties": {
      "name": {
        "type": "text"
      },
      "city": {
        "type": "keyword"
      }
    }
  }
}

PUT /test-c
{
  "mappings": {
    "properties": {
      "name": {
        "type": "text"
      },
      "phone": {
        "type": "keyword"
      }
    }
  }
}

POST /test-a/_doc/1
{
  "name": "John Doe 1",
  "email": "john@example.com"
}
POST /test-b/_doc/1
{
  "name": "John Doe 2",
  "city": "Buenos Aires"
}
POST /test-c/_doc/1
{
  "name": "John Doe 3",
  "phone": "+3106666666"
}
```
</details>

<details>
<summary>Configuration step</summary>

![Screenshot 2023-09-05 at 10 43 15](https://github.com/elastic/kibana/assets/1191206/745cb5df-71ee-4c0e-b935-8a02a3fab1d8)
![Screenshot 2023-09-05 at 10 43 31](https://github.com/elastic/kibana/assets/1191206/5ede0ac8-c02a-4a27-9c31-e559721bcb09)
![Screenshot 2023-09-05 at 10 43 55](https://github.com/elastic/kibana/assets/1191206/d7767c75-df65-4afb-813e-44402f8eb494)


</details>

<details>
<summary>Fields selection step</summary>

![Screenshot 2023-09-05 at 10 44 38](https://github.com/elastic/kibana/assets/1191206/0618317d-a139-4b10-be93-2d4f9e5ae9e4)
![Screenshot 2023-09-05 at 10 44 55](https://github.com/elastic/kibana/assets/1191206/519ddaee-e0d6-4d1e-aea5-991a9cc490bb)
![Screenshot 2023-09-05 at 10 45 09](https://github.com/elastic/kibana/assets/1191206/aa142631-ab24-47b9-a3e2-b6b4c3e3141e)


</details>

<details>
<summary>Summary step</summary>

![Screenshot 2023-09-05 at 10 47 09](https://github.com/elastic/kibana/assets/1191206/30b929b0-fb7d-467d-a1a7-b205d4dd0806)
![Screenshot 2023-09-05 at 10 47 43](https://github.com/elastic/kibana/assets/1191206/3cac5659-fcd1-4cfa-8803-9a351e4d8c60)



</details>

<details>
<summary>Error states</summary>
<img width="1434" alt="Screenshot 2023-08-25 at 09 10 53" src="https://github.com/elastic/kibana/assets/1191206/ff68ecb2-ad0f-4a40-b30d-ab4aecb1462b">


![Screenshot 2023-09-05 at 10 59 41](https://github.com/elastic/kibana/assets/1191206/0c1f43b6-09f3-4568-bf7f-069bc147ab92)


</details>
